src/components/TimePicker.js

```diff
@@ -0,0 +1,482 @@
+import PropTypes from 'prop-types';
+import React, {useCallback, useEffect, useMemo, useRef, useState} from 'react';
+import {View} from 'react-native';
+import _ from 'underscore';
+import useKeyboardShortcut from '@hooks/useKeyboardShortcut';
+import useLocalize from '@hooks/useLocalize';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import DateUtils from '@libs/DateUtils';
+import * as DeviceCapabilities from '@libs/DeviceCapabilities';
+import * as ValidationUtils from '@libs/ValidationUtils';
+import styles from '@styles/styles';
+import * as StyleUtils from '@styles/StyleUtils';
+import CONST from '@src/CONST';
+import AmountTextInput from './AmountTextInput';
+import BigNumberPad from './BigNumberPad';
+import Button from './Button';
+import FormHelpMessage from './FormHelpMessage';
+import refPropTypes from './refPropTypes';
+import Text from './Text';
+
+const propTypes = {
+    /** Refs forwarded to the TextInputWithCurrencySymbol */
+    forwardedRef: refPropTypes,
+
+    /** Default value for the inputs */
+    defaultValue: PropTypes.string,
+
+    /** Callback to call when the Save button is pressed  */
+    onSubmit: PropTypes.func.isRequired,
+
+    /** Callback to call when the input changes */
+    onInputChange: PropTypes.func,
+};
+
+const defaultProps = {
+    forwardedRef: null,
+    onInputChange: () => {},
+    defaultValue: '',
+};
+
+const AMOUNT_VIEW_ID = 'amountView';
+const NUM_PAD_CONTAINER_VIEW_ID = 'numPadContainerView';
+const NUM_PAD_VIEW_ID = 'numPadView';
+
+function insertAtPosition(originalString, newSubstring, selectionPositionFrom, selectionPositionTo, fn) {
```

Comment:
can we change `fn` to something that is more descriptive

Reply:
I changed this validation and simplified it, now it has comments.